app/src/main/java/com/google/android/samples/socialite/ui/camera/CameraViewModel.kt

```diff
@@ -343,14 +491,73 @@ class CameraViewModel @Inject constructor(
     }
 
     fun setZoomScale(scale: Float) {
-        val zoomState = camera?.cameraInfo?.zoomState?.value
+        val zoomState = camera.cameraInfo.zoomState.value
         if (zoomState == null) return
         val finalScale =
             (zoomState.zoomRatio * scale).coerceIn(
                 zoomState.minZoomRatio,
                 zoomState.maxZoomRatio,
             )
-        camera?.cameraControl?.setZoomRatio(finalScale)
+        camera.cameraControl.setZoomRatio(finalScale)
+    }
+
+    inner class SelfieSegmentationAnalyzer : ImageAnalysis.Analyzer {
+
+        val options = SelfieSegmenterOptions.Builder()
+            .setDetectorMode(SelfieSegmenterOptions.STREAM_MODE)
+            .build()
+        val selfieSegmenter = Segmentation.getClient(options)
+        var isAnalyzing = false
```

Comment:
I'm not sure if you need this `isAnalyzing` variable. Since you're using `ImageAnalysis.STRATEGY_KEEP_ONLY_LATEST`, you shouldn't be receiving any additional images until you've closed the previous image. See https://developer.android.com/reference/androidx/camera/core/ImageAnalysis#STRATEGY_KEEP_ONLY_LATEST()

Reply:
Done